catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java

```diff
@@ -0,0 +1,215 @@
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.suggestion;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.codice.ddf.spatial.geocoding.Suggestion;
+import org.codice.usng4j.CoordinateSystemTranslator;
+import org.codice.usng4j.DecimalDegreesCoordinate;
+import org.codice.usng4j.UsngCoordinate;
+import org.codice.usng4j.UtmUpsCoordinate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A regex-based processor for identifying potential UTM/UPS coordinate literals in a given input
+ * string despite deviations from the official spec. The general policy is to be as forgiving as
+ * possible without sacrificing determinism.
+ */
+public class UtmUpsCoordinateProcessor {
+  private static final Logger LOGGER = LoggerFactory.getLogger(UtmUpsCoordinateProcessor.class);
+
+  private static final Character WHITESPACE_CHAR = ' ';
+
+  private static final Pattern PATTERN_UTM_COORDINATE =
+      // TODO Test the 00H case
+      Pattern.compile(
+          "("
+              + UsngCoordinate.ZONE_REGEX_STRING
+              + ")"
+              + UsngCoordinate.LATITUDE_BAND_PART_ONE_REGEX_STRING
+              + "?(\\h)+(\\d){1,6}(me)?(\\h)+(\\d{1,7}|10000000)(mn)?",
+          Pattern.CASE_INSENSITIVE);
+
+  private static final Pattern PATTERN_UPS_COORDINATE =
+      Pattern.compile(
+          "0?[ABYZ](\\h)+(\\d){1,7}(me)?(\\h)+(\\d){1,7}(mn)?", Pattern.CASE_INSENSITIVE);
+
+  private static final Pattern PATTERN_UTM_OR_UPS_COORDINATE =
+      Pattern.compile(
+          String.format(
+              "(%s)|(%s)", PATTERN_UTM_COORDINATE.pattern(), PATTERN_UPS_COORDINATE.pattern()),
+          Pattern.CASE_INSENSITIVE);
+
+  // This key tells the UI that the geo is on the suggestion itself
+  private static final String LITERAL_SUGGESTION_ID = "LITERAL-UTM-UPS";
+
+  private final CoordinateSystemTranslator translator;
+
+  public UtmUpsCoordinateProcessor(CoordinateSystemTranslator translator) {
+    this.translator = translator;
+  }
+
+  /**
+   * Given a list of {@link Suggestion}s and the query that yielded them, enhance the list with
+   * additional suggestions based upon the presence of coordinate literals in the query string.
+   *
+   * @param results the list of results to enhance.
+   * @param query the query string to search for coordinate literals.
```

Comment:
is this a wkt or something else?  `query` is somewhat vague

Reply:
Updated to be more precise: 
```
@param query the user-provided gazetteer query text, which may contain coordinate literals.
```